lib/components/base_component.dart

```diff
@@ -104,27 +114,25 @@ abstract class BaseComponent extends Component {
   List<ComponentEffect> get effects => _effectsHandler.effects;
 
   /// Uses the game passed in, or uses the game from [HasGameRef] otherwise,
-
   /// to prepare the child component before it is added to the list of children.
   /// Note that this component needs to be added to the game first if
   /// [this.gameRef] should be used to prepare the child.
   /// For children that don't need preparation from the game instance can
   /// disregard both the options given above.
-  Future<void> addChild(Component c, {Game gameRef}) async {
-    assert(
-      gameRef != null || this is HasGameRef,
-      "Need gameRef either as an argument or from the HasGameRef mixin",
-    );
+  Future<void> addChild(Component child, {Game gameRef}) async {
     gameRef ??= (this as HasGameRef).gameRef;
     if (gameRef is BaseGame) {
-      gameRef.prepare(c);
+      gameRef.prepare(child);
     }
 
-    final childOnLoadFuture = c.onLoad();
+    final childOnLoadFuture = child.onLoad();
     if (childOnLoadFuture != null) {
       await childOnLoadFuture;
     }
-    _children.add(c);
+    _children.add(child);
+    if (gameRef?.components?.contains(this) ?? false) {
```

Comment:
I am wandering, how come this can be null? Maybe only when the mixin is not used?

Reply:
When the child is added before the parent is added to the game it will be null.

Reply:
But is that a behavior that we want to allow? Maybe we could add an assert to make sure that the parent already exists in the game before children can be added?

Reply:
I did have an assert that checked that, but since the parent -> child relationship is so general now I think we can allow it not to need to be prepared by the game? It is also mentioned in the comment on line 116.

Reply:
yeah, I think it could. I was worried that the onMount/onLoad on children would never be called, but I think those would end up been called when the parent is added to the game?

Reply:
Exactly, they would

Reply:
Fair enough then